README.md

```diff
@@ -1,16 +1,439 @@
+record-oriented application built, we will have a shared context to understand
+these challenging concepts in JavaScript (there's a pun in there, but you won't
+get it until the end of this lesson).
+
+## Define Execution Context
+
+When a function in JavaScript ***is called***, it is provided an _execution
+context_.
+
+The _execution context_ is a JavaScript `Object` that is either implicitly or
+explicitly passed along with the function at the time of the function's call.
+
+The implicit way of passing a context with a function is something we have to
+memorize and accept as part of the nature of JavaScript.
+
+The tools for explicitly passing a context at function call-time are the
+methods `call`, `apply`, and `bind.`
+
+## Define `this`
+
+The JavaScript keyword `this` returns the current _execution context_. Whether
+that context was passed explicitly or implicitly, `this` returns it.
+
+We'll now identify places where the _execution context_ is implicitly set.
+
+### Access Implicitly-Set Global Object in a JavaScript Engine
+
+```javascript
+this //=> Window
+typeof this //=> "object"
+this === window //=> true
+```
+
+In a brand new Chrome console, try the code above. All the code you write in a
+JavaScript file can be thought of as "one big function" whose context is set to
+the `Window` object. Thus `this === window` is `true`.
+
+We can try the same thing out in NodeJS. If you have `node` installed on your
+computer, open the Node REPL by typing `node`.
+
+```js
+this === global //=> true
+```
+
+Node calls the default _execution context_ `global`, but otherwise it's a
+global object like `window`.
+
+> **STICKY PROBLEM**: Keen-eyed readers might be thinking: "But is `this ===
+> Window` also `true`?" It is not. The `window` property is part of the
+> `Window` global object; yes, the `Window` `Object` has a property that points
+> to itself.
+>
+> Due to the strictness of `===`, a property pointing to a thing is not the
+> same thing as the thing itself. As such `this === Window` is `false`, but `this
+> === window` is `true` _despite the fact_ that they both ultimately point to
+> the same thing.
+>
+> If all this strikes you as a bit fussy and uninteresting, it's not essential
+> understanding for the rest of this module. For more on this consult
+> [Window.window][w1].
+
+### Access Implicitly-Set Global Object in a Function Call
+
+***All function declarations and function expressions*** you have called to
+this point have also had an _execution context_, you just didn't know to ask
+for it with `this`.
+
+```js
+let contextReturner = function() {
+  return this
+}
+
+contextReturner() //=> window
+contextReturner() === window //=> true
+```
+
+When you invoked a function, the global context was _implicitly_ passed.
+
+JavaScript does this so that by default our functions can access any properties
+on that most-global object, `window` (or `global` in Node). Consider:
```

Comment:
Is that the reasoning? I assumed that they decided that functions should have a context, and chose the global object as a default because they made a mistake. You can always access `window` or `global` from inside a function, so making it _also_ accessible as `this` doesn't seem like it helps with making it more accessible

Reply:
silent dot here on global obj

Reply:
tie to speak ex